geode-core/src/main/java/org/apache/geode/internal/statistics/StatisticsRegistry.java

```diff
@@ -104,6 +108,13 @@ public StatisticsRegistry(String name, long startTime) {
     this.osStatisticsFactory = osStatisticsFactory;
     this.atomicStatisticsFactory = atomicStatisticsFactory;
     this.pidSupplier = pidSupplier;
+    meterWhitelist = createMeterWhitelist();
+  }
+
+  private static Set<String> createMeterWhitelist() {
+    Set<String> meterWhitelist = new HashSet<>();
+    meterWhitelist.addAll(GatewayReceiverMeters.whitelist());
```

Comment:
Why allocate a new set here? If you must then inline this not the constructor so the initial allocation can account for capacity. Or consider returning an immutable wrapped reference to the original.

Reply:
I like returning an immutable wrapped reference. Assuming we don't need to update it later.

Reply:
While this should be infrequently called we still want to be mindful of waste. If we can avoid allocating heap great. As long as this method is not in any critical path we should be ok. I believe this method is only called when initially establishing the meter and not for every call to the meter right?

Reply:
Just once when initializing the StatisticsRegistry. I think that means once per `InternalDistributedSystem`.